config/default/tox.ini.j2

```diff
@@ -0,0 +1,35 @@
+[tox]
+envlist =
+    format
+    lint
+
+[testenv]
+py_files = find %(dist_path)s -name "*.py"
+rst_files = find %(dist_path)s -name "*.rst"
+allowlist_externals =
+    sh
+
+[testenv:format]
+description = automatically reformat python code
+skip_install = true
+deps =
+    pyupgrade
+    isort
+    black
+    -c lint-requirements.txt
+commands =
+    sh -c '{[testenv]py_files} | xargs pyupgrade --py38-plus *.py'
+    sh -c '{[testenv]py_files} | xargs isort *.py'
+    sh -c '{[testenv]py_files} | xargs black *.py'
+
+[testenv:lint]
+description = run linters that will help improve the code style
+skip_install = true
+deps =
+    flake8
+    codespell
+    -c lint-requirements.txt
+commands =
+    sh -c '{[testenv]py_files} | xargs flake8 *.py'
+    sh -c '{[testenv]py_files} | xargs codespell *.py'
+    sh -c '{[testenv]rst_files} | xargs codespell *.rst'
```

Comment:
If practical, I would like to use a single tool for spell checking. We use Vale for documentation in `plone.api`, `plone.restapi`, `volto`, `documentation`, and `training` repos. It adds style guide checking as well as spell checking, both of which are configurable. `make vale` invokes it.

Anyway, let's check Markdown, too.

```suggestion
    sh -c '{[testenv]rst_files} | xargs codespell *.rst'
    sh -c '{[testenv]md_files} | xargs codespell *.md'
```

Reply:
We can switch to [vale](https://pypi.org/project/vale/) no problem 👍🏾 does it need any configuration, or it runs on its own? 🤔 I never used it 😅 

As for markdown files, yes, good idea! 🍀

Reply:
@stevepiercy on [plone.api](https://github.com/plone/plone.api) I found no reference to `vale` and on [plone.restapi](https://github.com/plone/plone.restapi) only on the `Makefile`, but nothing on `.github/workflows` nor anywhere where it mentions to install it.

Even more confusing, sorry 😓 , is that on the `Makefile` it does `vale sync` and then `vale --no-wrap $FILES`, but the first `vale sync` already fails as there is no `.vale.ini` 🤔 

Am I missing something?

Reply:
ok, I got it running, I copied the `.vale.ini` from [documentation](https://github.com/plone/documentation) repository 😄 

Now, after installing `vale` and `rst2html` I got quite a lot of errors, which I'm not sure they all make sense...

```
 plone/batching/batching.rst
 19:14   error       Did you really mean 'iterable'?             Vale.Spelling
 68:32   error       Did you really mean 'items_on_page'?        Vale.Spelling
 125:70  error       Did you really mean 'pagesize'?             Vale.Spelling           
 142:23  error       Did you really mean 'navlist'?              Vale.Spelling           
 150:21  error       Did you really mean 'navlist'?              Vale.Spelling
```

Trying to add backticks on `navlist` still makes it an error, but if it's a technical word that one is meant to read, specially in the context of `pone.batching` because it is an attribute of an object... how one is meant to ignore mark this terms as ignored? 🤔

Reply:
As I mentioned on discord, getting `vale` to work on a per repository base seems a bit too much, at least from my initial understanding (my first day actually looking and using `vale`).

Maybe we can keep `codespell` for now, to not block this PR longer, and during Alpine City Sprint we have a look together @stevepiercy?

Reply:
Indeed. Vale may be better suited to repos with pages of docs, or where autodoc scans docstrings of many modules.

As far as the errors reported, we currently ignore them because there are far too many to manage at the moment, and we have gaping holes in the docs that have a much higher priority. Any spell checker out of the box will spew lots of errors. It has no clue about many technical words that are used in Plone, and we need to add them to the accept list.